src/librustc/hir/lowering.rs

```diff
@@ -3283,8 +3283,10 @@ impl<'a> LoweringContext<'a> {
             }
             // This is the normal case.
             AnonymousLifetimeMode::PassThrough => self.new_implicit_lifetime(span),
-
-            AnonymousLifetimeMode::ReportError => self.new_error_lifetime(None, span),
+            // elide the lifetime parameter (if possible) for the lifetime resolver to check
```

Comment:
It's not clear to me from this comment why this change works. Could you try to elaborate what this does in your comment?